### PR TITLE
Rename local variable `arguments` to `args`

### PR DESCRIPTION
**Pull request by andy-ms**
This may be affecting performance? CC @rbuckton

### CODE REVIEW

**Comment by RyanCavanaugh** [comment]
Any measurements?

File: src/compiler/utilities.ts [hunk 1]
```diff
@@ -1411,16 +1411,16 @@ namespace ts {
         if (callExpression.kind !== SyntaxKind.CallExpression) {
             return false;
         }
-        const { expression, arguments } = callExpression as CallExpression;
+        const { expression, arguments: args } = callExpression as CallExpression;
```
Comment (yuit) [inline src/compiler/utilities.ts]: Would it be better to just rename property in CallExpression?

Comment (rbuckton) [inline src/compiler/utilities.ts]: That would be a breaking change to API consumers.

**Comment by rbuckton** [comment]
@RyanCavanaugh there was a spike in the build-to-build performance tests associated with this change. Whether this was the direct cause I haven't yet determined. Regardless of the performance metrics, [we should *never* shadow `arguments` with a local or reassign it](https://github.com/petkaantonov/bluebird/wiki/Optimization-killers#33-assignment-to-arguments). It deoptimizes the function in v8, and is an error in strict mode.

**Comment by andy-ms** [comment]
The PR that added this code was #14984 on April 3rd. The performance tester showed emit time spike after that commit, but this function is called by the binder and checker. In the time since, emit time has gone back down.
That link doesn't mention shadowing, just mutation. Can you verify that deoptimization is actually taking place?

**Comment by rbuckton** [comment]
@andy-ms the initial `arguments` binding is hoisted inside of a function. A "redeclaration" of `arguments` using `var` doesn't really do anything, as can be observed in node:
```
> function f() { var arguments; console.log(arguments.length); }
> f(); 
0
> f("a");
1
```

So any assignment to `arguments`, even with a variable initializer, is actually reassigning the initial `arguments` binding.

That said, I ran a javascript project through tsc with hydrogen tracing and didn't see any specific deoptimizations in this function. I still think its worth the change.

**Comment by andy-ms** [comment]
Ah, forgot we're compiling to ES5.

**Pull request merged**